stackslib/src/net/api/get_tenures_fork_info.rs

```diff
@@ -0,0 +1,361 @@
+        chainstate: &StacksChainState,
+    ) -> Result<Self, ChainError> {
+        let first_block_mined = if !sn.sortition {
+            None
+        } else {
+            // is this a nakamoto sortition?
+            let epoch = SortitionDB::get_stacks_epoch(sortdb.conn(), sn.block_height)?.ok_or_else(
+                || {
+                    warn!(
+                        "Failed to lookup stacks epoch for processed snapshot height {}",
+                        sn.block_height
+                    );
+                    ChainError::InvalidChainstateDB
+                },
+            )?;
+            if epoch.epoch_id < StacksEpochId::Epoch30 {
+                StacksChainState::get_stacks_block_header_info_by_consensus_hash(
+                    chainstate.db(),
+                    &sn.consensus_hash,
+                )?
+                .map(|header| header.index_block_hash())
+            } else {
+                NakamotoChainState::get_nakamoto_tenure_start_block_header(
+                    chainstate.db(),
+                    &sn.consensus_hash,
+                )?
+                .map(|header| header.index_block_hash())
+            }
+        };
+        Ok(TenureForkingInfo {
+            burn_block_hash: sn.burn_header_hash.clone(),
+            burn_block_height: sn.block_height,
+            sortition_id: sn.sortition_id.clone(),
+            parent_sortition_id: sn.parent_sortition_id.clone(),
+            consensus_hash: sn.consensus_hash.clone(),
+            was_sortition: sn.sortition,
+            first_block_mined,
+        })
+    }
+}
+
+impl RPCRequestHandler for GetTenuresForkInfo {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.start_sortition = None;
+        self.stop_sortition = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let result = node.with_node_state(|_network, sortdb, chainstate, _mempool, _rpc_args| {
+            let start_from = self
+                .stop_sortition
+                .clone()
+                .ok_or_else(|| ChainError::NoSuchBlockError)?;
+            let recurse_end = self
+                .start_sortition
+                .clone()
+                .ok_or_else(|| ChainError::NoSuchBlockError)?;
+            let mut results = vec![];
+            let mut cursor = SortitionDB::get_block_snapshot_consensus(sortdb.conn(), &start_from)?
+                .ok_or_else(|| ChainError::NoSuchBlockError)?;
+            results.push(TenureForkingInfo::from_snapshot(
+                &cursor, sortdb, chainstate,
+            )?);
+            let handle = sortdb.index_handle(&cursor.sortition_id);
+            let mut depth = 0;
+            while depth < DEPTH_LIMIT && cursor.consensus_hash != recurse_end {
```

Comment:
I'm going to insist that you stream this back, instead of doing up to `DEPTH_LIMIT` loads.  The stream implementation would load one sortition on each pass.

Reply:
Letting the client just paginate on this endpoint is simpler for now, switching to a stream implementation could be done in follow on work.